Neos.Flow/Classes/Http/Component/SecurityEntryPointComponent.php

```diff
@@ -39,19 +40,36 @@ public function handle(ComponentContext $componentContext)
         }
 
         $actionRequest = $componentContext->getParameter(DispatchComponent::class, 'actionRequest');
-        $entryPoint = array_reduce($this->securityContext->getAuthenticationTokens(), function ($foundEntryPoint, TokenInterface $currentToken) {
-            return $foundEntryPoint ?? $currentToken->getAuthenticationEntryPoint();
-        }, null);
-
-        if ($entryPoint === null) {
+        $firstTokenWithEntryPoint = $this->getFirstTokenWithEntryPoint();
+        if ($firstTokenWithEntryPoint === null) {
             $this->securityLogger->notice('No authentication entry point found for active tokens, therefore cannot authenticate or redirect to authentication automatically.');
             throw $authenticationException;
         }
 
+        $entryPoint = $firstTokenWithEntryPoint->getAuthenticationEntryPoint();
         $this->securityLogger->info(sprintf('Starting authentication with entry point of type "%s"', get_class($entryPoint)), LogEnvironment::fromMethodName(__METHOD__));
-        $this->securityContext->setInterceptedRequest($actionRequest->getMainRequest());
+
+        // TODO: We should only prevent storage of intercepted request in the session here, but we don't have a different storage mechanism yet.
+        if (!$firstTokenWithEntryPoint instanceof SessionlessTokenInterface) {
+            $this->securityContext->setInterceptedRequest($actionRequest->getMainRequest());
+        }
+
         /** @var ResponseInterface $response */
         $response = $entryPoint->startAuthentication($componentContext->getHttpRequest(), $componentContext->getHttpResponse());
         $componentContext->replaceHttpResponse($response);
     }
+
+    /**
+     * Returns the first authenticated token that has an Authentication Entry Point configured, or NULL if none exists
+     *
+     * @return TokenInterface|null
+     */
+    private function getFirstTokenWithEntryPoint(): ?TokenInterface
+    {
+        foreach ($this->securityContext->getAuthenticationTokens() as $token) {
+            if ($token->getAuthenticationEntryPoint() !== null) {
+                return $token;
+            }
+        }
```

Comment:
Missing return statement, says psalm and I second ;)

Reply:
picky bastards :)
This was just a test to see if the checks work *g